testing/integration/otel_test.go

```diff
@@ -0,0 +1,295 @@
+
+service:
+  pipelines:
+    logs:
+      receivers: [filelog]
+      processors: [resource]
+      exporters:
+        - debug
+        - otlp/elastic`
+
+func TestFileProcessing(t *testing.T) {
+	define.Require(t, define.Requirements{
+		Group: Default,
+		Local: true,
+		OS: []define.OS{
+			// input path missing on windows
+			{Type: define.Linux},
+			{Type: define.Darwin},
+		},
+	})
+
+	t.Cleanup(func() {
+		_ = os.Remove(fileProcessingFilename)
+	})
+
+	fixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+	err = fixture.Prepare(ctx, fakeComponent, fakeShipper)
+	require.NoError(t, err)
+
+	// replace default elastic-agent.yml with otel config
+	// otel mode should be detected automatically
+	err = fixture.Configure(ctx, fileProcessingConfig)
+	require.NoError(t, err)
+
+	var fixtureWg sync.WaitGroup
+	fixtureWg.Add(1)
+	go func() {
+		defer fixtureWg.Done()
+		err = fixture.RunWithClient(ctx, false)
+	}()
+
+	require.Eventually(t,
+		func() bool {
+			// verify file exists
+			content, err := os.ReadFile(fileProcessingFilename)
+			return err == nil && len(content) > 0
+		},
+		5*time.Minute, 500*time.Millisecond,
+		"there should be exported logs by now")
+
+	cancel()
+	fixtureWg.Wait()
+	require.True(t, err == nil || err == context.Canceled || err == context.DeadlineExceeded, "Retrieved unexpected error: %s", err.Error())
+}
+
+func TestAPMIngestion(t *testing.T) {
+	info := define.Require(t, define.Requirements{
+		Group: Default,
+		Stack: &define.Stack{},
+		Local: true,
+		OS: []define.OS{
+			// apm server not supported on darwin
+			{Type: define.Linux},
+		},
+	})
+
+	// prepare agent
+	fixture, err := define.NewFixture(t, define.Version())
+	require.NoError(t, err)
+
+	ctx, cancel := testcontext.WithDeadline(t, context.Background(), time.Now().Add(10*time.Minute))
+	defer cancel()
+	err = fixture.Prepare(ctx, fakeComponent, fakeShipper)
+	require.NoError(t, err)
+
+	// prepare input
+	agentWorkDir := fixture.WorkDir()
+	fileName := "content.log"
+	fixture.WriteFileToWorkDir(ctx, "", fileName)
+
+	testUuid, err := uuid.NewV4()
+	require.NoError(t, err, "failed to create test id")
+	testId := testUuid.String()
+
+	apmConfig := fmt.Sprintf(apmOtelConfig, filepath.Join(agentWorkDir, fileName), testId)
+
+	err = fixture.Configure(ctx, []byte(apmConfig))
+	require.NoError(t, err)
+
+	componentsDir, err := aTesting.FindComponentsDir(agentWorkDir)
+	require.NoError(t, err)
+
+	// start apm default config just configure ES output
+	esHost, err := getESHost()
+	require.NoError(t, err, "failed to get ES host")
+	require.True(t, len(esHost) > 0)
+
+	esUsername := os.Getenv("ELASTICSEARCH_USERNAME")
+	require.True(t, len(esUsername) > 0)
+
+	esPass := os.Getenv("ELASTICSEARCH_PASSWORD")
+	require.True(t, len(esPass) > 0)
+
+	esClient := info.ESClient
+	if esClient == nil {
```

Comment:
Why do we need this check?  Under what circumstances would `info.ESClient` not be set in this test?

Reply:
i had this for use cases when commented out using Stack and used my own

Reply:
Since we are using `Stack: &define.Stack{},` now, can we remove this section and the `getESClient` function?  None of the other integration tests have something like this so I don't think we should introduce this usage pattern here. 

If you want to use your own stack while debugging this test, you should be able to keep `Stack: &define.Stack{},` uncommented and run the test like so (this is what I do):

```
sudo ELASTICSEARCH_USERNAME=xxxx ELASTICSEARCH_PASSWORD=xxxxxxxxxxxx ELASTICSEARCH_HOST=xxxxxxxxxxx TEST_DEFINE_PREFIX=foo  go test -tags integration -v ./testing/integration/... -test.run TestAPMIngestion -count 1
```